packages/components/src/box-control/utils.ts

```diff
@@ -7,7 +7,42 @@ import { __ } from '@wordpress/i18n';
  * Internal dependencies
  */
 import { parseQuantityAndUnitFromRawValue } from '../unit-control/utils';
-import type { BoxControlProps, BoxControlValue } from './types';
+import type {
+	BoxControlProps,
+	BoxControlValue,
+	CustomValueUnits,
+} from './types';
+
+export const CUSTOM_VALUE_SETTINGS: CustomValueUnits = {
+	px: { max: 300, steps: 1 },
+	'%': { max: 100, steps: 1 },
+	vw: { max: 100, steps: 1 },
+	vh: { max: 100, steps: 1 },
+	em: { max: 10, steps: 0.1 },
+	rm: { max: 10, steps: 0.1 },
+	svw: { max: 100, steps: 1 },
+	lvw: { max: 100, steps: 1 },
+	dvw: { max: 100, steps: 1 },
+	svh: { max: 100, steps: 1 },
+	lvh: { max: 100, steps: 1 },
+	dvh: { max: 100, steps: 1 },
+	vi: { max: 100, steps: 1 },
+	svi: { max: 100, steps: 1 },
+	lvi: { max: 100, steps: 1 },
+	dvi: { max: 100, steps: 1 },
+	vb: { max: 100, steps: 1 },
+	svb: { max: 100, steps: 1 },
+	lvb: { max: 100, steps: 1 },
+	dvb: { max: 100, steps: 1 },
+	vmin: { max: 100, steps: 1 },
+	svmin: { max: 100, steps: 1 },
+	lvmin: { max: 100, steps: 1 },
+	dvmin: { max: 100, steps: 1 },
+	vmax: { max: 100, steps: 1 },
+	svmax: { max: 100, steps: 1 },
+	lvmax: { max: 100, steps: 1 },
+	dvmax: { max: 100, steps: 1 },
+};
 
 export const LABELS = {
 	all: __( 'All' ),
```

Comment:
I feel like it would be a bit confusing to a screen reader user without the word "sides". Same with "horizontal" and "vertical" 🤔 It might be clearer and more translation-friendly if we straight up said "Top and bottom sides" and "Left and right sides". Or at least "Horizontal sides"?

```suggestion
	all: __( 'All sides' ),
```

Reply:
Definitely, great suggestion! I think "Top and bottom sides" and "Left and right sides" makes the most sense.

Reply:
I think it would also make sense to update 'Mixed' as well. Do you think 'Mixed sides' is clear?

Reply:
Thinking about [this placeholder behavior](https://github.com/WordPress/gutenberg/pull/56665#discussion_r1446076146) made me realize that "Mixed" is mixed _values_, not mixed _sides_. Given the limited space for the placeholder, it might be better to just keep it "Mixed", and maybe add a translators note in case that helps.

`/* translators: Shown as input placeholder when there are mixed values. */` or something like that.